share-extension/PostActivity.m

```diff
@@ -0,0 +1,125 @@
+//
+// Copyright (C) 2003-2015 eXo Platform SAS.
+//
+// This is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License as
+// published by the Free Software Foundation; either version 3 of
+// the License, or (at your option) any later version.
+//
+// This software is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this software; if not, write to the Free
+// Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+// 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+//
+
+
+#import "PostActivity.h"
+
+@implementation PostActivity
+
+
+#pragma mark - Doc activity
+/*
+ Make the file upload name on server side
+ fileAttacheName = mobile + [datetime string] + [converted filename]
+ converted rules:
+ - No upper case
+ - No special characters
+ - No space or -,_
+ */
+#define REGEXP_SEPARATER @"[_: ]"
+#define REGEXP_2TIME_SPEPARATER @"-{2,}"
+#define REGEXP_REMOVE_CHAR @"[{}()!@#$%^&|;\"~`'<>?\\/,+=*.ˆ]"
```

Comment:
Add more forbidden chars: 
- [ and ]
- cf the regexp on Android: [`!@#\$%\^&\*\|;"'<>/\\\[\]\{\}\(\)\?,=\+\.]+